File: CoderBytes/py/exam-longestword.py
def longestword(s: str) -> str:
    """
    Trouve le mot le plus long contenant uniquement
    des lettres et des chiffres.
    """
    max_length = 0
    longest = ''
    
    for word in s.split(" "):
        # Vérifier si le mot contient seulement des lettres et chiffres
        if word.isalnum():
            word_length = len(word)
            if word_length > max_length:
                max_length = word_length
                longest = word
    
    return longest


# Tests
print(longestword("fun!!& time"))      # time
print(longestword("I love dogs"))      # love

File: CoderBytes/php/exam-longestword.php
<?php
function longestword(string $str){
    $wordCount = 0;
    $word = '';
    foreach(explode(" ", $str) as $c){
        if (preg_match("/^[a-zA-Z0-9]+$/",$c)){
            $wlen = strlen($c);
            if ($wlen >$wordCount){
                $wordCount = $wlen;
                $word = $c; 
            }
        }
    }
    return $word;
}


echo longestword("fun!!& time")."\n";  // time
echo longestword("I love dogs")."\n";  // love